crates/iota-indexer/src/models/transactions.rs

```diff
@@ -196,6 +196,12 @@ impl StoredTransaction {
         self.events.get(idx).cloned().flatten()
     }
 
+    /// True for checkpointed transactions, False for optimistically indexed
+    /// transactions
+    pub fn is_checkpointed_transaction(&self) -> bool {
+        self.checkpoint_sequence_number >= 0
+    }
```

Comment:
For optimistically indexed transaction it will be -1?

Reply:
Then, I would suggest doing the check directly on `-1`, which seems more precise:

```
    pub fn is_checkpointed_transaction(&self) -> bool {
        self.checkpoint_sequence_number != -1
    }
```

Reply:
Okay then it's fine as it is! Thanks for clarification.